frontend/src/lib/utils.tsx

```diff
@@ -463,17 +464,38 @@ export const humanFriendlyMilliseconds = (timestamp: number | undefined): string
 
     return `${(timestamp / 1000).toFixed(2)}s`
 }
-export function humanFriendlyDuration(d: string | number | null | undefined, maxUnits?: number): string {
+
+function customRound(value: number, decimalPlaces: number | undefined): number {
+    const factor = Math.pow(10, decimalPlaces ?? 0)
+    return Math.round(value * factor) / factor
+}
+
+/**
+ * Converts a number of seconds to a human-readable duration string.
+ * constructs days, hours, minutes, seconds
+ * returns either the number of days and hours (when over 1 day)
+ * or the number of hours, minutes, and seconds (when under 1 day)
+ *
+ * @param d the number of seconds to convert
+ * @param maxUnits the maximum number of units to display (from the left) (e.g. 2 would display only hours and minutes)
+ * @param decimalPlaces is applied to the right most value this is most useful when dealing with small number of seconds e.g. to distinguish 1.2 and 1.4 seconds instead of showing 1s for both
+ */
+export function humanFriendlyDuration(
+    d: string | number | null | undefined,
+    maxUnits?: number,
+    decimalPlaces?: number
```

Comment:
can set `DEFAULT_DECIMAL_PLACES` over here?
Guessing if this is `undefined`, this function errors out?